src/app/client/src/app/modules/workspace/components/content-creation-default-template/content-creation-default-template.component.ts

```diff
@@ -0,0 +1,304 @@
+        });
+      }
+    });
+    if (callback) {
+      callback(associations);
+    }
+  }
+
+  /**
+* @description                    - Which is used to resolve the dependency.
+* @param {Object} field           - Which field need need to get change.
+* @param {Object} associations    - Association values of the respective field.
+* @param {Boolean} resetSelected  - @default true Which defines while resolving the dependency dropdown
+*                                   Should reset the selected values of the field or not
+*/
+  applyDependencyRules(field, associations, resetSelected) {
+    // reset the depended field first
+    // Update the depended field with associated value
+    // Currently, supported only for the dropdown values
+    let dependedValues;
+    let groupdFields;
+    if (field.depends && field.depends.length) {
+      _.forEach(field.depends, (id) => {
+        if (resetSelected) {
+          this.resetSelectedField(id, field.range);
+        }
+        dependedValues =  _.map(associations, i => _.pick(i, ['name', 'category']));
+        groupdFields = _.chain(dependedValues)
+          .groupBy('category')
+          .map((name, category) => ({ name, category }))
+          .value();
+        this.updateDropDownList(id, dependedValues);
+        if (groupdFields.length) {
+          _.forEach(groupdFields, (value, key) => {
+            this.updateDropDownList(value.category, _.map(value.name, i => _.pick(i, 'name')));
+          });
+        } else {
+          this.updateDropDownList(id, []);
+        }
+      });
+    }
+  }
+
+  /**
+   * @description         - Which is used to restore the dropdown slected value.
+   * @param {String} id   - To restore the specific dropdown field value
+   */
+  resetSelectedField(id, field) {
+    this.formInputData[id] = '';
+  }
+
+
+  /**
+* @description            - Which updates the drop down value list
+*                         - If the specified values are empty then drop down will get update with master list
+* @param {Object} field   - Field which is need to update.
+* @param {Object} values  - Values for the field
+*/
+  updateDropDownList(fieldCode, values) {
+    if (values.length) {
+      this.categoryList[fieldCode] = _.unionBy(values, 'name');
+    } else {
+      this.mapMasterCategoryList(this.formFieldProperties, fieldCode);
+    }
+  }
+
+  /**
+*
+* @description                     -
+* @param {Object} configurations   - Field configurations
+* @param {String} key              - Field uniq code
+*/
+  mapMasterCategoryList(configurations, key) {
+    _.forEach(configurations, (field, value) => {
+      if (key) {
+        if (field.code === key) {
+          this.categoryList[field.code] = field.range;
+        }
+      } else {
+        if (field.range) {
+          this.categoryList[field.code] = field.range;
+        }
+      }
+    });
+  }
+  /**
+   * @method getYearsForCreateTextBook
+   */
+  getYearsForCreateTextBook() {
```

Comment:
use sementic ui datepicker with datetype as year. Dont use this function to display year in number